provider/v2/azure/common/common.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+
+	"github.com/openclarity/vmclarity/provider"
+)
+
+func HandleAzureRequestError(err error, actionTmpl string, parts ...interface{}) (bool, error) {
```

Comment:
note: this return obscures what is being returned when writing business code. I think this could be split into two helpers `HandleAzureRequestError` that only returns err, and `HandleAzureRequestStatusCodeError` that returns HTTP status code (its easier and much clearer to do a check against `http.StatusNotFound`) and the error. We can do this in another PR for all providers, but I am not finding it easy to understand what is happening in the implementation part when dealing with these error helpers.